systems/data-plan/api-gateway/pkg/rest/router.go

```diff
@@ -0,0 +1,276 @@
+	}
+
+	return resp, nil
+}
+func (p *Router) getBaseRatesHandler(c *gin.Context) (*pbBaseRate.GetBaseRatesResponse, error) {
+	provider := c.Query("provider")
+	effectiveAt := c.Query("effective_at")
+	simType := c.Query("sim_type")
+	country, ok := c.GetQuery("country")
+	if !ok {
+		return nil, &rest.HttpError{HttpCode: http.StatusBadRequest,
+			Message: "country is a mandatory query parameter"}
+	}
+	to, err := strconv.ParseUint(c.Param("to"), 10, 64)
+	if err != nil {
+		logrus.Error(err)
+		return nil, &rest.HttpError{HttpCode: http.StatusBadRequest,
+			Message: "to query param is not valid!"}
+	}
+	from, err := strconv.ParseUint(c.Param("from"), 10, 64)
+	if err != nil {
+		logrus.Error(err)
+		return nil, &rest.HttpError{HttpCode: http.StatusBadRequest,
+			Message: "from query param is not valid!"}
+	}
+
+	resp, err := p.clients.d.GetBaseRates(&pbBaseRate.GetBaseRatesRequest{
+		Country:     country,
+		Provider:    provider,
+		To:          to,
+		From:        from,
+		SimType:     pbBaseRate.SimType(pbBaseRate.SimType_value[simType]),
+		EffectiveAt: effectiveAt,
+	})
+	if err != nil {
+		logrus.Error(err)
+		return nil, err
+	}
+
+	return resp, nil
+}
+func (p *Router) getPackageHandler(c *gin.Context) (*pb.GetPackagesResponse, error) {
+	packageId, error := strconv.ParseUint(c.Param("package"), 10, 64)
+	if error != nil {
+		logrus.Error(error)
+		return nil, &rest.HttpError{HttpCode: http.StatusBadRequest,
+			Message: "packageId is not valid!"}
+	}
+	resp, err := p.clients.d.GetPackage(&pb.GetPackagesRequest{
+		Id: packageId,
+	})
+	if err != nil {
+		logrus.Error(err)
+		return nil, err
+	}
+
+	return resp, nil
+}
+func (p *Router) deletePackageHandler(c *gin.Context) (*pb.DeletePackageResponse, error) {
+	packageId, error := strconv.ParseUint(c.Param("package"), 10, 64)
+	if error != nil {
+		logrus.Error(error)
+		return nil, &rest.HttpError{HttpCode: http.StatusBadRequest,
+			Message: "packageId is not valid!"}
+	}
+	resp, err := p.clients.d.DeletePackage(&pb.DeletePackageRequest{
+		Id: packageId,
+	})
+	if err != nil {
+		logrus.Error(err)
+		c.JSON(http.StatusNotFound, gin.H{"error": err})
+		return nil, err
+		
+	}
+	return resp, nil
+}
+func (p *Router) UpdatePackageHandler(c *gin.Context, req *UpdatePackageRequest) (*pb.UpdatePackageResponse, error) {
+	packageId, error := strconv.ParseUint(c.Param("package"), 10, 64)
+	if error != nil {
+		logrus.Error(error)
+		return nil, &rest.HttpError{HttpCode: http.StatusBadRequest,
+			Message: "packageId is not valid!"}
+	}
+	resp, err := p.clients.d.UpdatePackage(&pb.UpdatePackageRequest{
+		Id:packageId,
+		Name:        req.Name,
+		SimType:     pb.SimType(pb.SimType_value[req.SimType]),
+		Active:      req.Active,
+		Duration:    req.Duration,
+		SmsVolume:   req.SmsVolume,
+		DataVolume:  req.DataVolume,
+		VoiceVolume: req.VoiceVolume,
+		OrgRatesId:  req.OrgRatesId,
+	})
+	if err != nil {
+		logrus.Error(err)
+		c.JSON(http.StatusNotFound, gin.H{"error": err})
```

Comment:
@Brackleycassinga   if you fail to update a package, then this is an internal server error.